eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs

```diff
@@ -270,15 +287,18 @@ instance NFData CostModel where
 
 -- | Convert cost model parameters to a cost model, making use of the
 --  conversion function mkEvaluationContext from the Plutus API.
-mkCostModel :: Language -> Map Text Integer -> Either CostModelApplyError CostModel
-mkCostModel PlutusV1 cm =
-  case PV1.mkEvaluationContext cm of
-    Right evalCtx -> Right (CostModel PlutusV1 cm evalCtx)
-    Left e -> Left e
-mkCostModel PlutusV2 cm =
-  case PV2.mkEvaluationContext cm of
-    Right evalCtx -> Right (CostModel PlutusV2 cm evalCtx)
+mkCostModel :: Language -> Map Text Integer -> Either PV1.CostModelApplyError CostModel
+mkCostModel lang cm =
+  case eCostModel of
+    Right (evalCtx, _) -> Right (CostModel lang cm evalCtx)
     Left e -> Left e
+  where
+    mkEvaluationContext =
+      case lang of
+        PlutusV1 -> PV1.mkEvaluationContext
+        PlutusV2 -> PV2.mkEvaluationContext
+    eCostModel :: Either PV1.CostModelApplyError (PV1.EvaluationContext, [CostModelApplyWarn])
+    eCostModel = runWriterT (mkEvaluationContext (Map.elems cm))
```

Comment:
wonderful! it's awesome to see the plan put into action! https://github.com/input-output-hk/cardano-ledger/issues/2902#issuecomment-1183644628

There is actually now no longer a need to keep the keys in the cost model. In other words, we can replace the `Map Text Integer` with `[Int]`.